src/js/components/Button/Button.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Button
+ *   https://teamsnap-ui-patterns.netlify.com/patterns/components/button.html
+ *
+ * A common button component that will render the appropriate styles for a button or a button-link. Check out the
+ * teamsnap-ui patterns library for more styling info.
+ *
+ * Example:
+ *  <Button
+ *    text="Click Me"
+ *    handleClick={ () => console.warn('Clicky Clicky') }
+ *    attributes={{ className="Button Button--primary", onFocus: () => console.warn('FOCUSED') }} />
+ *
+ */
+
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import TextLink from '../TextLink'
+import { getClassName } from '../../utils/helpers'
+
+class Button extends PureComponent {
+  getButtonClassName = () => {
+    const { className, color, size, isActive, mods } = this.props
+
+    return getClassName(
+      className,
+      color && `Button--${color}`,
+      size && `Button--${size}`,
+      isActive && "is-active",
+      mods
+    )
+  }
+
+
+  renderButtonLink = () => {
+    const { children, routerLink, location, style, isDisabled, onClick } = this.props
+
+    return (
+      <TextLink
+        className={ this.getButtonClassName() }
+        children={ children }
+        routerLink={ routerLink }
+        location={ location }
+        style={ style }
+        disabled={ isDisabled }
+        onClick={ onClick } />
+    )
+  }
+
+  renderButton = () => {
+    const { children, type, style, isDisabled, onClick } = this.props
+
+    return (
+      <button
+        type={ type }
+        className={ this.getButtonClassName() }
+        style={ style }
+        onClick={ onClick }
+        disabled={ isDisabled }
+        children={ children } />
+    )
+  }
+
+  render () {
+    const { location } = this.props
+    return location ? this.renderButtonLink() : this.renderButton()
+  }
+}
+
+Button.propTypes = {
+  children: PropTypes.node.isRequired,
```

Comment:
I'm not sure this should always be required, since a button will not have children per se. It will have text. Unless I'm thinking of this wrong. An href could hold a node or text.

Reply:
If button only accepts children as say the text or icon, then I think it's safe to have required here.

Reply:
```
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
```
What if you want to send in an array of nodes?
```
<Button>
 <icon>
 Text
</Button>
```

Reply:
I don't really think that is necessary.  I think just accepting children should be enough for these buttons, right?